File: common/src/main/java/com/ultreon/mods/advanceddebug/client/menu/pages/ItemEntityPage.java
package com.ultreon.mods.advanceddebug.client.menu.pages;

import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext;
import com.ultreon.mods.advanceddebug.util.TargetUtils;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.world.entity.item.ItemEntity;
import net.minecraft.world.phys.EntityHitResult;
import org.jetbrains.annotations.NotNull;

import static net.minecraft.ChatFormatting.RED;

public class ItemEntityPage extends EntityPage {
    public ItemEntityPage() {
    }

    @Override
    public void render(@NotNull GuiGraphics gfx, IDebugRenderContext ctx) {
        EntityHitResult entityHit = TargetUtils.entity();
        if (entityHit != null && entityHit.getEntity() instanceof ItemEntity entity) {
            ctx.left("Generic");
            ctx.left("Owner", entity.getOwner());
            ctx.left("Item", entity.getItem());
            ctx.left("Spin", entity.getSpin(minecraft.getFrameTime()));
            ctx.left("Age", entity.getAge());
            ctx.left();

            ctx.right("Flags");
            ctx.right("Attackable", entity.isAttackable());
            ctx.right();
        } else {
            // not looking at a block, or too far away from one to tell
            ctx.top(RED + "<No Item Entity Was Found>");
        }
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/client/menu/pages/ItemEntityPage.kt
package com.ultreon.mods.advanceddebug.client.menu.pages

import com.mojang.blaze3d.vertex.PoseStack
import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext
import com.ultreon.mods.advanceddebug.util.TargetUtils
import net.minecraft.ChatFormatting
import net.minecraft.world.entity.item.ItemEntity

class ItemEntityPage(modId: String, name: String) : EntityPage(modId, name) {
    override fun render(poseStack: PoseStack, ctx: IDebugRenderContext) {
        val entityHit = TargetUtils.entity()
        val entity = entityHit?.entity
        if (entityHit != null && entity is ItemEntity) {
            ctx.left("Generic")
            ctx.left("Thrower", entity.thrower)
            ctx.left("Owner", entity.owner)
            ctx.left("Item", entity.item)
            ctx.left()
            ctx.right("Flags")
            ctx.right("Attackable", entity.isAttackable)
            ctx.right()
        } else {
            // not looking at a block, or too far away from one to tell
            ctx.top(ChatFormatting.RED.toString() + "<No Item Entity Was Found>")
        }
    }
}